packages/loopring_v3/contracts/impl/auxiliaries/DowntimeCostCalculator.sol

```diff
@@ -0,0 +1,150 @@
+import "../../lib/MathUint.sol";
+
+import "../../iface/IDowntimeCostCalculator.sol";
+
+
+/// @title The default IDowntimeCostCalculator implementation.
+/// @author Daniel Wang  - <daniel@loopring.org>
+contract DowntimeCostCalculator is IDowntimeCostCalculator, Claimable
+{
+    using MathUint for uint;
+
+    uint public basePrice;
+    uint public maxPenalty;
+    uint public gracePeriods;
+    uint public gracePeriodPrice;
+    uint public maxAwailableDowntime;
+
+    event SettingsUpdated(
+        uint oldBasePrice,
+        uint oldMaxPenalty,
+        uint oldGracePeriodMinutes,
+        uint oldGracePeriodPrice,
+        uint oldMaxAwailableDowntime
+    );
+
+    constructor(
+        uint _basePrice,
+        uint _maxPenalty,
+        uint _gracePeriods,
+        uint _gracePeriodPrice,
+        uint _maxAwailableDowntime
+        )
+        Claimable()
+        public
+    {
+        updateSettings(
+            _basePrice,
+            _maxPenalty,
+            _gracePeriods,
+            _gracePeriodPrice,
+            _maxAwailableDowntime
+        );
+    }
+
+    function getDowntimeCostLRC(
+        uint  totalTimeInMaintanance,
+        uint  totalLifetime,
+        uint  awailableDowntime,
+        uint  /* exchangeStakedLRC */,
+        uint  downtimeToPurchase
+        )
+        external
+        view
+        returns (uint)
+    {
+        uint newCost = getTotalCost(
+            totalTimeInMaintanance,
+            totalLifetime,
+            awailableDowntime.add(downtimeToPurchase)
+        );
+
+        uint oldCost = getTotalCost(
+            totalTimeInMaintanance,
+            totalLifetime,
+            awailableDowntime
+        );
+
+        return newCost > oldCost ? newCost - oldCost : 0;
+    }
+
+    function updateSettings(
+        uint _basePrice,
+        uint _maxPenalty,
+        uint _gracePeriods,
+        uint _gracePeriodPrice,
+        uint _maxAwailableDowntime
+        )
+        public
+        onlyOwner
+    {
+        require(
+            _basePrice > 0 && _maxPenalty > 0 &&
+            _gracePeriodPrice > 0 && _maxAwailableDowntime > 0,
+            "ZERO_VALUE"
+        );
+        require(_gracePeriodPrice <= _basePrice, "INVALID_PRICE");
+
+        emit SettingsUpdated(
+            basePrice,
+            maxPenalty,
+            gracePeriods,
+            gracePeriodPrice,
+            maxAwailableDowntime
+        );
+
+        basePrice = _basePrice;
+        maxPenalty = _maxPenalty;
+        gracePeriods = _gracePeriods;
+        gracePeriodPrice = _gracePeriodPrice;
+        maxAwailableDowntime = _maxAwailableDowntime;
+    }
+
+    function getTotalCost(
+        uint totalTimeInMaintanance,
+        uint totalLifetime,
+        uint downtime
+        )
+        private
+        view
+        returns (uint)
+    {
+        require(downtime <= maxAwailableDowntime, "PURCHASE_PROHIBITED");
+        uint total = totalTimeInMaintanance.add(downtime);
```

Comment:
`totalTimeInMaintanance` is in seconds, but `downtime` is in minutes here.

Reply:
Good catch.

Reply:
fixed.